_posts/2021-10-27-g2p-contributing.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title:  "Contributing a new mapping to g2p for everyone to use"
+author: AidanPine
+categories: []
+tags: [ tutorial, explanation, tech-showcase, g2p ]
+description: "Tutorial: Contributing a new mapping to g2p for everyone to use"
+featured: false
+hidden: false
+---
+
+This is the last part of the seven-part series on g2p. In this part, we'll discuss how to contribute your mappings to the main g2p library.
+
+### G2P Blog Series Index
+
+- [Background]({{ "g2p-background" | absolute_url }})
+- [How to write a basic mapping in G2P Studio]({{ "g2p-basic-mappings-gui" | absolute_url }})
+- [Writing mappings on your computer]({{ "g2p-basic-mappings-local" | absolute_url }})
+- [Advanced mappings]({{ "g2p-advanced-mappings" | absolute_url }})
+- [ReadAlong Studio & Other Applications]({{ "g2p-applications" | absolute_url }})
+- [Preprocessing mappings]({{ "g2p-preprocess" | absolute_url }})
+- Contributing
+
+## Advanced: contributing your rules to the main `g2p` library
+
+So, you've written some cool rules and you want to contribute, that's awesome! There are lots of benefits to contributing your mapping to `g2p`. First of all, once your mapping is accepted, you'll have it available and live on [G2P Studio](https://g2p-studio.herokuapp.com). Second, once the next version of `g2p` is released with your mapping, it will be automatically built in to the [Convertextract]({{ "convertextract-app" | absolute_url }}) library. Third, if your mapping is between a language's writing system and the IPA, you can also get [ReadAlongs]({{ "g2p-applications" | absolute_url }}) support for your language.
+
+So, you write your mapping once, and you get three things for free (G2P studio, convertextract and readalongs). Here's how:
+
+1. Fork g2p, see <https://docs.github.com/en/github/getting-started-with-github/fork-a-repo> for more details
+2. Add a folder for your language using the appropriate [ISO 639.3 code](https://en.wikipedia.org/wiki/List_of_ISO_639-3_codes) to `g2p/mappings/langs`
+3. Add a `config.yaml` file as described [here]({{ "g2p-basic-mappings" | absolute_url }})
+4. Add your mapping in that same folder (`g2p/mappings/langs/<yourlang>`)
```

Comment:
if you accepted my suggestion at 2 and 3, you don't need to put the parenthesized example here anymore.

I find the whole thing clearer if the example is given at the folder creation instruction.